config/config.go

```diff
@@ -166,14 +223,327 @@ func (cs configs) ResolveConfig(f func(configs []interface{}) error) error {
+	if o.Credentials == nil {
+		return nil, false, nil
+	}
+
+	return o.Credentials, true, nil
+}
+
+// WithCredentialsProvider is a helper function to construct functional options
+// that sets Credential provider value on config's LoadOptions
+func WithCredentialsProvider(v aws.CredentialsProvider) func(*LoadOptions) error {
+	return func(o *LoadOptions) error {
+		o.Credentials = v
+		return nil
+	}
+}
+
+// getProcessCredentialOptions returns the wrapped function to set processcreds.Options
+func (o LoadOptions) getProcessCredentialOptions(ctx context.Context) (func(*processcreds.Options), bool, error) {
+	if o.ProcessCredentialOptions == nil {
+		return func(*processcreds.Options) {}, false, nil
+	}
+
+	return o.ProcessCredentialOptions, true, nil
+}
+
+// WithProcessCredentialOptions is a helper function to construct functional options
+// that sets a function to use processcreds.Options on config's LoadOptions
+func WithProcessCredentialOptions(v func(*processcreds.Options)) func(*LoadOptions) error {
+	return func(o *LoadOptions) error {
+		o.ProcessCredentialOptions = v
+		return nil
+	}
+}
+
+// getEC2RoleCredentialOptions returns the wrapped function to set the ec2rolecreds.Options
+func (o LoadOptions) getEC2RoleCredentialOptions(ctx context.Context) (func(*ec2rolecreds.Options), bool, error) {
+	if o.EC2RoleCredentialOptions == nil {
+		return func(*ec2rolecreds.Options) {}, false, nil
+	}
+
+	return o.EC2RoleCredentialOptions, true, nil
+}
+
+// WithEC2RoleCredentialOptions is a helper function to construct functional options
+// that sets a function to use ec2rolecreds.Options on config's LoadOptions
+func WithEC2RoleCredentialOptions(v func(*ec2rolecreds.Options)) func(*LoadOptions) error {
+	return func(o *LoadOptions) error {
+		o.EC2RoleCredentialOptions = v
+		return nil
+	}
+}
+
+// getEndpointCredentialOptions returns the wrapped function to set endpointcreds.Options
+func (o LoadOptions) getEndpointCredentialOptions(context.Context) (func(*endpointcreds.Options), bool, error) {
+	if o.EndpointCredentialOptions == nil {
+		return func(*endpointcreds.Options) {}, false, nil
+	}
+
+	return o.EndpointCredentialOptions, true, nil
+}
+
+// WithEndpointCredentialOptions is a helper function to construct functional options
+// // that sets a function to use endpointcreds.Options on config's LoadOptions
+func WithEndpointCredentialOptions(v func(*endpointcreds.Options)) func(*LoadOptions) error {
+	return func(o *LoadOptions) error {
+		o.EndpointCredentialOptions = v
+		return nil
+	}
+}
+
+// getWebIdentityRoleCredentialOptions returns the wrapped function
+func (o LoadOptions) getWebIdentityRoleCredentialOptions(context.Context) (func(*stscreds.WebIdentityRoleOptions), bool, error) {
+	if o.WebIdentityRoleCredentialOptions == nil {
+		return func(*stscreds.WebIdentityRoleOptions) {}, false, nil
+	}
+
+	return o.WebIdentityRoleCredentialOptions, true, nil
+}
```

Comment:
Nit inconsistency in these function if they are doing set vs not set as the first. Would be helpful for readability if they were all consistent.